app/models/miq_provision_virt_workflow.rb

```diff
@@ -185,6 +185,15 @@ def update_field_read_only(options = {})
     fields(:hardware) { |fn, f, _dn, _d| f[:read_only] = true if options[:read_only_fields].include?(fn) }
   end
 
+  def allowed_hosts_obj(options = {})
+    all_hosts = super
+    vlan_name = get_value(@values[:vlan])
+    return all_hosts unless vlan_name
+
+    _log.info "Filtering hosts with the following network: <#{vlan_name}>"
+    all_hosts.reject { |h| !h.lans.pluck(:name).include?(vlan_name) }
```

Comment:
Openstack infra is only for managing the undercloud - registering new baremetal nodes, adding more compute nodes, etc. In other words, there is no provisioning for Openstack Infra.